cpp/src/reductions/reduction_dispatcher.cuh

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CUDF_REDUCTION_DISPATCHER_CUH
+#define CUDF_REDUCTION_DISPATCHER_CUH
+
+#include "reduction_functions.cuh"
+
+namespace cudf {
+namespace reductions {
```

Comment:
I think this should be `namespace reduction`, not `reductions`, as Jake suggested.

Reply:
renamed to `reduction`

Reply:
shall I rename it `cudf::reduce()`?

Reply:
Rename the function cudf::reduction to cudf::reduce() to make it a verb like other APIs. Leave the namespace `cudf::reduction` to be a noun like other namespaces.